app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt

```diff
@@ -40,11 +40,20 @@ class LoginPresenter @Inject constructor(
         }
         val user = Login(email = email, password = password)
         if (!loginError) {
-            view?.showLoginLoading()
+            view?.toggleProgressBarPresence(true)
             networkRequest(userRepository.loginUser(user)) {
-                onResponseSuccessful { _ -> view?.logInUser() }
-                onResponseFailed { _, _ -> view?.showLoginError() }
-                onCallFailure { view?.showConnectionError() }
+                onResponseSuccessful {
+                    view?.logInUser()
+                    view?.toggleProgressBarPresence(false)
```

Comment:
move `view?.toggleProgressBarPresence(false)` to te bottom of the `networkRequest` method.